mc2mc/internal/client/odps.go

```diff
@@ -33,12 +33,23 @@ func NewODPSClient(logger *slog.Logger, client *odps.Odps) *odpsClient {
 // ExecSQL executes the given query in syncronous mode (blocking)
 // with capability to do graceful shutdown by terminating task instance
 // when context is cancelled.
-func (c *odpsClient) ExecSQL(ctx context.Context, query string) error {
+func (c *odpsClient) ExecSQL(ctx context.Context, query string, queryHints ...map[string]string) error {
 	if c.isDryRun {
 		c.logger.Info("dry run mode, skipping execution")
 		return nil
 	}
+
 	hints := addHints(c.additionalHints, query)
+
+	// add job-specific hints, which takes priority over the global hints
+	logHints := []string{}
```

Comment:
to avoid scattered additional hints, can we move all hints into one place?
seems like it's better to remove `Setup.SetupAdditionalHints` in `mc2mc.go`, and just put the `cfg.AdditionalHints` when try to execute the query `err := c.Execute(ctx, queryToExecute)`

and closure for injecting the id can still be same like this

Reply:
then there won't be any additional hints set in the client level, right?

Reply:
yes, correct